Tests/SonarScanner.MSBuild.PreProcessor.Test/SonarCloudWebServerTest.cs

```diff
@@ -312,21 +376,28 @@ private static IDownloader MockIDownloader(string cacheBaseUrl = null)
             return mock.Object;
         }
 
-        private static Mock<HttpMessageHandler> MockHttpHandler(bool cacheEnabled, string fullCacheUrl, string ephemeralCacheUrl, string token, Stream cacheData = null)
+        private static Mock<HttpMessageHandler> MockHttpHandler(string cacheFullUrl, string prepareReadResponse, HttpStatusCode prepareReadResponseCode = HttpStatusCode.OK)
         {
-            var mock = new Mock<HttpMessageHandler>();
-            mock.Protected()
+            var handler = new Mock<HttpMessageHandler>();
+            handler.Protected()
                 .Setup<Task<HttpResponseMessage>>(
                     "SendAsync",
-                    ItExpr.Is<HttpRequestMessage>(x => x.RequestUri == new Uri(fullCacheUrl) && x.Headers.Any(h => h.Key == "Authorization" && h.Value.Contains($"Bearer {token}"))),
+                    ItExpr.Is<HttpRequestMessage>(x => x.RequestUri == new Uri(cacheFullUrl) && x.Headers.Any(h => h.Key == "Authorization" && h.Value.Contains($"Bearer {Token}"))),
                     ItExpr.IsAny<CancellationToken>())
                 .Returns(Task.FromResult(new HttpResponseMessage
                 {
-                    StatusCode = HttpStatusCode.OK,
-                    Content = new StringContent($"{{ \"enabled\": \"{cacheEnabled}\", \"url\":\"{ephemeralCacheUrl}\" }}"),
+                    StatusCode = prepareReadResponseCode,
+                    Content = new StringContent(prepareReadResponse),
                 }))
                 .Verifiable();
 
+            return handler;
+        }
+
+        private static Mock<HttpMessageHandler> MockHttpHandlerWithStream(string cacheFullUrl, string ephemeralCacheUrl, Stream cacheData = null)
```

Comment:
I would make the method a `MockHttpHandler` overload. It takes a stream as a parameter so we don't need to specify this in the method name.